### PR TITLE
Fix permanently delete workspace documents

### PR DESCRIPTION
**Pull request by elioschmutz**
Jira: https://4teamwork.atlassian.net/browse/CA-2122

In the PR https://github.com/4teamwork/opengever.core/pull/6567 we introduced the possibility to remove trashed workspace documents. It was a soft-delete, so the delete action was just using another workflow transition which ensures, that nobody can see the deleted document instead managers.

In the PR https://github.com/4teamwork/opengever.core/pull/6620 we introduced a new placeful workflow for workspaces and a new workflow for workspace documents. This new workspace have no longer provided the transitions to soft-delete the documents. 

On the frontend-side, the delete action was still present, but deleting it was no longer possible:

https://user-images.githubusercontent.com/557005/116458007-6636de80-a864-11eb-9563-f8a0289f4ddc.mov

This PR fixes the broken delete action for workspace documents by introducing a new dedicated action for workspace documents and a new permission to be able to permanently delete workspace documents.

We don't want a soft-delete for workspace documents as it was before. So we don't add transitions to the workflow but added a new DELETE endpoint instead.

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- API change:
  - [ ] Documentation is updated
  - [ ] API Changelog entry (see [guide](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/451248812/API+Changelog+Guidelines))
  - If breaking:
    - [ ] api-change label added
    - [ ] #delivery channel notified about breaking change
    - [ ] Scrum master is informed
- New functionality:
  - [ ] for `document` also works for `mail`
  - [ ] for `task` also works for `forwarding`
- Further improvements needed:
  - [ ] Create follow-up stories and link them in the PR and Jira issue
- Upgrade steps (changes in profile):
  - [ ] Make it deferrable if possible
  - [ ] Execute as much as possible conditionally
- DB-Schema migration
  - [ ] All changes on a model (columns, etc) are included in a DB-schema migration.
  - [ ] Constraint names are shorter than 30 characters (`Oracle`)
- [ ] Change could impact client installations, client policies need to be adapted
- New translations
  - [ ] All msg-strings are unicode
- Change in schema definition:
  - [ ] If `missing_value` is specified, then `default` has to be set to the same value

### CODE REVIEW

File: opengever/api/tests/test_actions.py [hunk 1]
```diff
@@ -1338,6 +1351,75 @@ def test_actions_for_document_in_inactive_dossier(self, browser):
...
+        )
+
+    @browsing
+    def test_actions_for_trashed_document_in_workspace_folder(self, browser):
+        self.login(self.workspace_member, browser)
+        browser.open(self.workspace_folder_document.absolute_url() + '/@trash',
+                     method='POST', headers={'Accept': 'application/json'})
+
+        expected_object_buttons = [
+            {u'icon': u'', u'id': u'copy_item', u'title': u'Copy item'},
```
Comment (njohner) [inline opengever/api/tests/test_actions.py:1414]: huh, I don't think copy and move make sense for trashed documents? These actions are not available on the Gever side for trashed documents

File: opengever/core/profiles/default/actions.xml [hunk 2]
```diff
@@ -605,6 +605,20 @@
...
+    <object name="delete_workspace_document" meta_type="CMF Action" i18n:domain="plone">
+      <property name="title" i18n:translate="">Delete</property>
+      <property name="description" i18n:translate="" />
+      <property name="url_expr" />
+      <property name="link_target" />
+      <property name="icon_expr" />
+      <property name="available_expr" />
+      <property name="permissions">
+        <element value="opengever.workspace: Delete Documents" />
+        <element value="Remove GEVER content" />
```
Comment (njohner) [inline opengever/core/profiles/default/actions.xml:617]: Hum, I don't think that makes sense? A User with `Remove GEVER content` permission will not be able to use that action...

File: opengever/core/upgrades/20210428195416_add_workspace_document_delete_actions/actions.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    <object name="delete_workspace_document" meta_type="CMF Action" i18n:domain="plone">
+      <property name="title" i18n:translate="">Delete</property>
+      <property name="description" i18n:translate="" />
+      <property name="url_expr" />
+      <property name="link_target" />
+      <property name="icon_expr" />
+      <property name="available_expr" />
+      <property name="permissions">
+        <element value="opengever.workspace: Delete Documents" />
+        <element value="Remove GEVER content" />
```
Comment (njohner) [inline opengever/core/upgrades/20210428195416_add_workspace_document_delete_actions/actions.xml:14]: ✂️

File: opengever/core/profiles/default/actions.xml [hunk 4]
```diff
@@ -986,6 +986,19 @@
...
+    <object name="delete_workspace_document" meta_type="CMF Action" i18n:domain="plone">
+      <property name="title" i18n:translate="">Delete</property>
+      <property name="description" i18n:translate="" />
+      <property name="url_expr" />
+      <property name="link_target" />
+      <property name="icon_expr" />
+      <property name="available_expr">python:here.restrictedTraverse('@@plone_interface_info').provides('opengever.trash.trash.ITrashed')</property>
+      <property name="permissions">
+        <element value="opengever.workspace: Delete Documents" />
+        <element value="Remove GEVER content" />
```
Comment (njohner) [inline opengever/core/profiles/default/actions.xml:998]: Same here

File: opengever/core/upgrades/20210428195416_add_workspace_document_delete_actions/actions.xml [hunk 5]
```diff
@@ -18,4 +18,22 @@
...
+    <object name="delete_workspace_document" meta_type="CMF Action" i18n:domain="plone">
+      <property name="title" i18n:translate="">Delete</property>
+      <property name="description" i18n:translate="" />
+      <property name="url_expr" />
+      <property name="link_target" />
+      <property name="icon_expr" />
+      <property name="available_expr">python:here.restrictedTraverse('@@plone_interface_info').provides('opengever.trash.trash.ITrashed')</property>
+      <property name="permissions">
+        <element value="opengever.workspace: Delete Documents" />
+        <element value="Remove GEVER content" />
```
Comment (njohner) [inline opengever/core/upgrades/20210428195416_add_workspace_document_delete_actions/actions.xml:32]: ✂️

**Review by njohner — COMMENTED** [review]
@deiferni can you also have a quick look? 
This makes me nervous, because we introduce a permanent delete for documents, which has the potential to allow deleting documents also in Gever.
I'd definitely not allow that action for "Remove Gever Content" permission, restrict it to a permission that only exists on the Teamraum side. Apart from that it looks OK to me.

File: opengever/api/tests/test_actions.py (hunk 1 above)
Comment (njohner) [reply opengever/api/tests/test_actions.py:1414]: But that can be treated in a separate issue...

File: opengever/core/profiles/default/actions.xml (hunk 2 above)
Comment (elioschmutz) [reply opengever/core/profiles/default/actions.xml:617]: Right. I though, we need both, but it really makes no sense

**Comment by elioschmutz** [comment]
@njohner hum... the permission is only managed in the teamraum, so it should be safe... but probably your're right that it makes you nervous...

I could improve the `DeleteWorkspaceContent` class ,which is responsible for deleting workspace content, to only delete config within the `IWorkspaceRoot`. That would be really safe then... what do you think?

**Comment by elioschmutz** [comment]
@njohner @deiferni so what do you think? is it enough to extend the `DeleteWorkspaceContent` class with another check? Or do you have another idea?

**Review by deiferni — COMMENTED** [review]
@elioschmutz we have quickly discussed it and the check in `DeleteWorkspaceContent` would be nice.

Also while implementing please have a look at `ContentDelete` in `plone.restapi`. Maybe you can reuse it instead of copying. Something i just stumbled upon.

**Comment by elioschmutz** [comment]
@njohner @deiferni 👍 updated

**Review by njohner — APPROVED** [review]
LGTM

**Pull request merged**